### DISCUSSION TITLE
`withAuthUserTokenSSR` with `Cache-Control`

### DISCUSSION THREAD

**Discussion by MartinXPN** [gladly-team/next-firebase-auth#532] [Q&A] [answered]
I was wondering what would be the best way to cache some of the content generated on the server-side (like 3rd party API calls that don't depend on the user token, or static content) for pages that use `withAuthUserTokenSSR` (those are fetched inside `getServerSideProps` to render the page).

Are we restricted to using only the `private` `Cache-Control` header if we use `withAuthUserTokenSSR`?
Are there maybe some workarounds that we can use to cache the static content that should be accessible to all the users?

Amazon CloudFront offers an option to cache based on a specific cookie but I'm not sure if Vercel does that.

Any kind of suggestion would be really appreciated.

**Comment by kmjennison** [comment, marked as answer]
As you suggest, this is more about the hosting provider and Next.js than it is about `next-firebase-auth`. However, if this package is blocking some implementation of a solution, let me know.

More to your question: you could always use CloudFront or another CDN in front of Vercel, if needed.

**Reply by MartinXPN** [reply]
Thanks for the response @kmjennison ! If this makes the website too slow or overwhelms with too many users I'll try to use the CloudFront in front of Vercel (I think they discourage that to make sure the caches are invalidated properly when you deploy a new app on Vercel).
